pkg/cloudprovider/providers/aws/aws_loadbalancer.go

```diff
@@ -727,7 +706,26 @@ func (c *Cloud) updateInstanceSecurityGroupsForNLBTraffic(actualGroups []*ec2.Se
 			}
 
 		}
+
+		// MTU discovery
+		mtuRuleAnnotation := fmt.Sprintf("%s=%s", NLBMtuDiscoveryRuleDescription, lbName)
+		mtuPermission := &ec2.IpPermission{
+			IpProtocol: aws.String("icmp"),
+			FromPort:   aws.Int64(3),
+			ToPort:     aws.Int64(4),
+		}
+		ranges := []*ec2.IpRange{}
+		for _, cidr := range clientCidrs {
+			ranges = append(ranges, &ec2.IpRange{
+				CidrIp:      aws.String(cidr),
+				Description: aws.String(mtuRuleAnnotation),
+			})
+		}
+		mtuPermission.IpRanges = ranges
+
 		if len(adds) > 0 {
+			adds = append(adds, mtuPermission)
```

Comment:
This is not going to work, if a service port was changed from `80` to `8080`, you'd get both an add and remove.

The ICMP rules really need their own tracking structure, you could probably do something similar to the 
`portChanges` map:

```go
// securityGroupID, clientCIDR, add/remove
icmpPortChanges := map[string]map[string]bool{}
```

You'd need to
* get the desired state (which is just an ICMP for each clientCidr for each security group)
* get the actual state (probably just reuse the `for _, actualGroup := range actualGroups` loop)
* append to `adds`/ `removes` as necessary